test/wanda_web/auth/jwt_auth_plug_test.exs

```diff
@@ -0,0 +1,69 @@
+defmodule WandaWeb.Auth.JWTAuthPlugTest do
+  use WandaWeb.ConnCase, async: true
+
+  import Mox
+
+  alias WandaWeb.Auth.JWTAuthPlug
+
+  setup [:set_mox_from_context, :verify_on_exit!]
+
+  setup do
+    stub(
+      Joken.CurrentTime.Mock,
+      :current_time,
+      fn ->
+        1_671_715_992
+      end
+    )
+
+    :ok
+  end
+
+  describe "call/1" do
+    test "should return the connection with the user_id as subject" do
+      jwt =
+        "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJ0cmVudG8tcHJvamVjdCIsImV4cCI6MTY3MTcxNjU5MiwiaWF0IjoxNjcxNzE1OTkyLCJpc3MiOiJodHRwczovL2dpdGh1Yi5jb20vdHJlbnRvLXByb2plY3Qvd2ViIiwianRpIjoiMnN0bW9wOTY5b3RrOTNpNjU0MDAwYmViIiwibmJmIjoxNjcxNzE1OTkyLCJzdWIiOjEsInR5cCI6IkJlYXJlciJ9.KnUXDdH2IHd5hDJFrc-KXzXrRmWwotc26t9L2CksQs4"
+
+      conn =
+        build_conn()
+        |> put_req_header("authorization", "Bearer " <> jwt)
+        |> JWTAuthPlug.call(%{})
+
+      refute conn.halted
+      assert conn.private.user_id == 1
+    end
+
+    test "should  be halted and have unauthorized status if the token is expired" do
+      expired_jwt =
+        "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJ0cmVudG8tcHJvamVjdCIsImV4cCI6MTY3MTY0MjQxNCwiaWF0IjoxNjcxNjQxODE0LCJpc3MiOiJodHRwczovL2dpdGh1Yi5jb20vdHJlbnRvLXByb2plY3Qvd2ViIiwianRpIjoiMnNwaTFvbmxxbml1ZnE5dnVrMDAwMG9hIiwibmJmIjoxNjcxNjQxODE0LCJzdWIiOjEsInR5cCI6IkJlYXJlciJ9.oub6_NsHcVIyd0de14Lzk3SuCMMgr8O-sSWLr7Gxcp8"
```

Comment:
just as curiosity, how do you create this invalid keys? like this expired example

Reply:
you can use http://jwtbuilder.jamiekurtz.com/